lib/xenops_server.ml

```diff
@@ -932,6 +932,23 @@ let is_nvidia_sriov vgpus pci =
 
 let is_not_nvidia_sriov vgpus pci = not (is_nvidia_sriov vgpus pci)
 
+(* [must_dequarantine] identifies GPUs that need to be de-quarantined *)
+let must_dequarantine vgpu =
+  Xenops_interface.Vgpu.(match vgpu.implementation with
+  | GVT_g _ -> true
+  | Nvidia _ -> true
+  | _ -> false
+  )
+
+(* compute a list of PCI_dequarantine operations for PCI devices *)
+let dequarantine_ops vgpus pcis =
+  List.concat
+  [ vgpus |> List.filter must_dequarantine |> List.map
+        Xenops_interface.Vgpu.(fun vgpu -> PCI_dequarantine vgpu.physical_pci_address)
+  ; pcis  |> List.filter (is_nvidia_sriov vgpus) |> List.map
```

Comment:
And MXGPU?

Reply:
As far as I understand it does not need to be de-quarantined. Am i wrong?